src/runtime/analytics-service.js

```diff
@@ -138,14 +146,43 @@ export class AnalyticsService {
 
   /**
    * @return {!Promise}
+   * @private
+   */
+  init_() {
+    if (!this.serviceReady_) {
+      // TODO(sohanirao): Potentially do this even earlier
+      this.doc_.getBody().appendChild(this.getElement());
+      this.serviceReady_ = this.activityPorts_.openIframe(
+          this.iframe_, this.src_, this.args_).then(port => {
+            this.portResolver_(port);
+            return this.setContext_();
+          });
+    }
+    return this.serviceReady_;
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
    */
-  start() {
-    this.doc_.getBody().appendChild(this.getElement());
-    return this.activityPorts_.openIframe(this.iframe_, this.src_,
-        this.args_).then(port => {
-          this.portResolver_(port);
-          return this.setContext_();
+  start_() {
+    return this.init_().then(() => {
```

Comment:
The `start` and `init` and how they cooperate right now is very complicated, but I don't see why that should be the case. Correct me if I'm wrong, but this should be enough?

```
/** @private {!Promise<!ActivityPort>} */
this.serviceReady_ = null;

start_() {
  if (!this.serviceReady_) {
    this.doc_.getBody().appendChild(this.getElement());
    this.serviceReady_ = this.activityPorts_.openIframe(...)
        .then(port => {
            this.setContext_();
            return port.whenReady().then(() => port);
        });
  }
  return this.serviceReady_;
}

logEvent(event) {
  this.stary_().then(port => {
    port.message({...});
  });
}
```

So, the other vars/methods `port_`, `portResolver_`, `portReady_`, etc - all not needed...